### PR TITLE
Recon implementation

### LINKED ISSUE #1029: Data Reconciliation for Validation framework
Once we release a new version of the overwatch this module will ensure whether the data is consistent across the current release and previous release. 
In order to perform the data reconciliation, we need two overwatch deployments with current and previous versions having the same configurations(primordial date, max days) and then check whether the data between deployments is matching or not. This validation check should be configurable to include/exclude tables or even the columns specific to tables. We should also be able to configure the number of days to be considered for comparison.

After running the reconciliation it will generate a report. The report will contain for each table

Total number of rows
Total number of rows matching
Total number of rows not matching
ETC

### PR DESCRIPTION
**Pull request by sriram251-code**
closes #1029

### CODE REVIEW

File: src/main/scala/com/databricks/labs/overwatch/validation/DataReconciliation.scala [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+        val msg = "Got exception while reading from pipeline_report ,"
+        println(msg+e.getMessage)
+        throw e
+    }
+
+  }
+
+
+
+}
```
Comment (souravbaner-da) [inline src/main/scala/com/databricks/labs/overwatch/validation/DataReconciliation.scala:236]: @sriram251-code  Please resolve the merge conflict

File: src/main/scala/com/databricks/labs/overwatch/validation/DataReconciliation.scala [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+  /**
+   * Function is the starting point of the reconciliation.
+   * @param sourceEtl :  ETL name of Previous version of OW
+   * @param targetEtl : ETL name of the current version of OW
+   */
+  private[overwatch] def performRecon(sourceEtl:String,targetEtl:String): Unit ={
+    val sourceOrgIDArr = getAllOrgID(sourceEtl)
+    val targetOrgIDArr = getAllOrgID(targetEtl)
+    performBasicValidation(sourceOrgIDArr,targetOrgIDArr)
+    val sourceWorkspace = getConfig(sourceEtl,targetOrgIDArr(0))
```
[the remark this answers is not in the document]
Comment (sriram251-code) [reply src/main/scala/com/databricks/labs/overwatch/validation/DataReconciliation.scala]: Its for getting the config and deriving all the targets it does not matter but still i have changed it

File: src/main/scala/com/databricks/labs/overwatch/validation/DataReconciliation.scala [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+    val targetWorkspace = getConfig(targetEtl,targetOrgIDArr(0))
+    val targets = getAllTargets(sourceWorkspace)
```
[the remark this answers is not in the document]
Comment (sriram251-code) [reply src/main/scala/com/databricks/labs/overwatch/validation/DataReconciliation.scala:39]: we already checking it on performBasicRecon() method

File: src/main/scala/com/databricks/labs/overwatch/validation/DataReconciliation.scala [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+      .mode("append")
+      .save(s"""${validationPath}/report/${reportName}""")
+    println("Validation report has been saved to " + s"""${validationPath}/report/${reportName}""")
+  }
+
+
+
+private def getQuery(tableName: String,orgId: String,target:PipelineTable): String = {
+  val _keys = target.keys.filterNot(_ == "Overwatch_RunID")
+  val reconColymns = target.reconColumns
```
[the remark this answers is not in the document]
Comment (sriram251-code) [reply src/main/scala/com/databricks/labs/overwatch/validation/DataReconciliation.scala]: done

File: src/main/scala/com/databricks/labs/overwatch/validation/DataReconciliation.scala (hunk 1 above)
Comment (sriram251-code) [reply src/main/scala/com/databricks/labs/overwatch/validation/DataReconciliation.scala:236]: done

**Comment by sriram251-code** [comment]
Duplicated

**Pull request closed without merging**